### PR TITLE
Sign Hash michelson data

### PR DESCRIPTION
**Pull request by jmininger**

### CODE REVIEW

File: APDUs.md [hunk 1]
```diff
@@ -82,7 +82,9 @@ The main difference between `INS_SIGN` and `INS_SIGN_UNSAFE` is that
 included in the APDU data. This is unsafe, because the user doesn’t
 see what operation they are actually signing. When this happens, we
 tell the user “Unrecognized: Sign Hash” so that they can make
-appropriate external steps to verify this hash.
+appropriate external steps to verify this hash. The difference between
+`INS_SIGN` and `INS_SIGN_WITH_HASH` is that the latter returns both the
+signature *AND* the hash of the data (while the former only returns the signature).
```
Comment (xplat) [inline APDUs.md:87]: These documentation updates reflect things that were already the case, and you are just doing a late update?

File: src/apdu_sign.c [hunk 2]
```diff
@@ -570,12 +570,21 @@ static size_t handle_apdu(bool const enable_hashing, bool const enable_parsing,
...
 	    if (G.packet_index == 1) {
+
 	        G.maybe_ops.is_valid = false;
-                G.magic_byte = get_magic_byte_or_throw(buff, buff_size);
-		parse_operations_init(&G.maybe_ops.v, G.key.derivation_type, &G.key.bip32_path, &G.parse_state);
+          G.magic_byte = get_magic_byte_or_throw(buff, buff_size);
+
+          // If it is an "operation" (starting with the 0x03 magic byte), set up parsing
+          // If it is arbitrary Michelson (starting with 0x05), dont bother parsing and show the "Sign Hash" prompt
+          if (G.magic_byte == MAGIC_BYTE_UNSAFE_OP) {
```
Comment (xplat) [inline src/apdu_sign.c:579]: What makes sure this starts with 0x05 if it doesn't start with 0x03?

File: APDUs.md (hunk 1 above)
Comment (jmininger) [reply APDUs.md:87]: Yeah, just trying to make things more clear for onboarding

File: src/apdu_sign.c (hunk 2 above)
Comment (jmininger) [reply src/apdu_sign.c:579]: The `get_magic_byte_or_throw()` function handles this, although i will add an extra check

**Pull request merged**